src/triage/component/postmodeling/utils/predictions_selected_model_groups.py

```diff
@@ -0,0 +1,175 @@
+        Args:
+            model_groups (List[int]): A list of model groups in the experiment
+            experiment_hashes (List[str]): Optional. A list of experiment hashes we are interested in.
+                If this is provided, only the model_ids that are relevant to the given experiments will be returned
+
+        Return: 
+            A DataFrame that contains the relevant model_ids, train_end_times, and matrix information
+    """
+
+    q = """
+        SELECT 
+            model_id,
+            train_end_time,
+            model_hash,
+            model_group_id,
+            train_matrix_uuid,
+            c.matrix_uuid as test_matrix_uuid
+        FROM triage_metadata.experiment_models a 
+            JOIN triage_metadata.models b using(model_hash)
+                JOIN test_results.prediction_metadata c using(model_id)
+        WHERE model_group_id in ({model_groups})"""
+
+    args_dict = {
+        'model_groups': ', '.join([str(x) for x in model_groups])
+    }
+
+    if experiment_hashes is not None:
+        q = q + " AND experiment_hash in ({experiment_hashes})"
+        args_dict['experiment_hashes'] = ', '.join(["'" + str(x) + "'" for x in experiment_hashes])
+    
+    q = q.format(**args_dict)
+
+    return pd.read_sql(q, db_engine)
+
+
+def _summary_of_models(models_df):
+    """generates logging summaries and warnings about the models being scored"""
+
+    # how many models per model group?
+    cts = models_df.groupby('model_group_id').count()['model_id']
+    cts.name = 'num_models'
+    model_counts = cts.unique()
+    if len(model_counts) > 1:
+        logging.warning('No. of models is different between model groups. See below: \n {}'.format(cts.reset_index()))
+    else:
+        logging.info('Each model group contains {} models'.format(model_counts[0]))
+
+    # how many models per model group at each train_end_time?
+    # Sometimes there are more than one model for each model group with different random seeds
+    time_cts = models_df.groupby(['train_end_time', 'model_group_id']).count()['model_id']
+    time_cts.name = 'num_models'
+    msk = time_cts > 1
+
+    if len(time_cts[msk]) > 1:
+        logging.warning('There are model groups with more than one model per train_end_time. See below: \n {}'.format(time_cts[msk].reset_index()))
+        
+    
+def generate_predictions(db_engine, model_groups, project_path, experiment_hashes=None, train_end_times_range=None, rank_order='worst', replace=False):
+    """ Generate predictions and write to DB for a set of model_groups in an experiment
+        Args:
+            db_conn: Sqlalchemy engine
+            model_groups (list): The list of model group ids we are interested in (ideally, chosen through audition)
+            project_path (str): Path where the created matrices and trained model objects are stored for the experiment
+            experiment_hashes (List[str]): Optional. hash of the experiment (currently handling only one experiment)
```

Comment:
Didn't you update it to all for more than one experiment?

Reply:
Yes, updated the docstring